tests/src/jit/CheckProjects/CheckProjects.cs

```diff
@@ -0,0 +1,427 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Xml.Serialization;
+using System.Text;
+using System.Xml.Linq;
+
+// Verify that jit test project files specify DebugType properly.
+// Returns error status (-1) if any project files are in error.
+
+internal class ScanProjectFiles
+{
+    private static bool s_showNeedsFixOnly = true;
+    private static bool s_tryAndFix = true;
+    private static int s_projCount = 0;
+    private static int s_needsFixCount = 0;
+    private static int s_deferredFixCount = 0;
+    private static int s_fixedCount = 0;
+
+    private static int Main(string[] args)
+    {
+        // If invoked w/o args, locate jit test project dir from
+        // CORE_ROOT, and scan only.
+        //
+        // If invoked w args, locate and try to fix project files.
+        string projectRoot = null;
+
+        if (args.Length == 0)
+        {
+            s_tryAndFix = false;
+
+            // CORE_ROOT should be something like
+            //  c:\repos\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root
+            // or
+            //  D:\j\workspace\x64_release_w---0575cb46\bin\tests\Windows_NT.x64.Release\Tests\Core_Root
+            // We want
+            //  c:\repos\coreclr\tests\src\JIT
+            string coreRoot = System.Environment.GetEnvironmentVariable("CORE_ROOT");
+
+            if (coreRoot == null)
+            {
+                Console.WriteLine("CORE_ROOT must be set");
+                return -1;
+            }
+
+            int binIndex = coreRoot.IndexOf("bin");
+
+            if (binIndex < 0)
+            {
+                Console.WriteLine("CORE_ROOT must be set to full path to repo test dir; was '{0}'.",
+                    coreRoot);
+                return -1;
+            }
+
+            string repoRoot = coreRoot.Substring(0, binIndex);
+
+            projectRoot = Path.Combine(repoRoot, "tests", "src", "JIT");
+        }
+        else if (args.Length != 1)
+        {
+            Console.WriteLine("Usage: scan <dir>");
```

Comment:
Nit: it would be nice if this message explained that when invoked with args, this tool will rewrite parts of your tree.